generator/main.ts

```diff
@@ -55,10 +63,15 @@ async function main([out_dir, ...idl_files]) {
       return index == 0 ? match.toUpperCase() : match;
     });
   });
-
+  let names = new Array<String>();
   for (let idl_file of idl_files) {
-    await generateInterface(env, idl_file, out_dir);
+    const parsedData = webidl.parse(await file.read(path.resolve(idl_file)));
+    const idl_fragments: idls.Fragments = new idls.Fragments(parsedData);
+    names.push(new String(idl_fragments.definitions[0].name));
```

Comment:
i think we need to check type before push to figure out whether it is interfaces or not.